File: PhotonAI_Agents/generated_output/price_calculator.py
"""Price calculator module for discount and tax calculations.

This module provides functionality for calculating discounted prices and applying
tax rates to products, following Google Python Style Guidelines.
"""

from __future__ import annotations

import sys
from typing import Union


class PriceCalculator:
    """Calculator for price operations including discounts and taxes."""

    @staticmethod
    def calculate_discounted_price(
        price: float, 
        discount_percentage: float
    ) -> float:
        """Calculates the price after applying a discount.

        Args:
            price: The original price of the item.
            discount_percentage: The discount percentage to apply (0-100).

        Returns:
            The discounted price.

        Raises:
            ValueError: If discount percentage is not between 0 and 100.
        """
        if discount_percentage < 0 or discount_percentage > 100:
            raise ValueError(
                "Discount percentage must be between 0 and 100."
            )
        return price - (price * discount_percentage / 100.0)

    @staticmethod
    def calculate_final_price_with_tax(price: float, tax_rate: float) -> float:
        """Calculates the final price after applying tax.

        Args:
            price: The base price before tax.
            tax_rate: The tax rate to apply (e.g., 0.05 for 5%).

        Returns:
            The final price including tax.

        Raises:
            ValueError: If tax rate is negative.
        """
        if tax_rate < 0:
            raise ValueError("Tax rate cannot be negative.")
        return price + (price * tax_rate)


def main() -> None:
    """Main function to run the price calculator interactively."""
    try:
        original_price = float(input("Enter the original price: "))
        
        discount = float(input("Enter the discount percentage: "))
        
        discounted_price = PriceCalculator.calculate_discounted_price(
            original_price, discount
        )
        print(f"Price after {discount:.2f}% discount: ₹{discounted_price:.2f}")
        
        tax = float(input("Enter the tax rate (e.g., 0.05 for 5%): "))
        
        final_price = PriceCalculator.calculate_final_price_with_tax(
            discounted_price, tax
        )
        print(f"Final price with tax: ₹{final_price:.2f}")
        
    except ValueError as e:
        print(f"Error: {e}", file=sys.stderr)
    except Exception as e:
        print(f"Unexpected error: {e}", file=sys.stderr)


if __name__ == "__main__":
    main()

File: PhotonAI_Agents/generated_output/PriceCalculator.java
import java.util.Scanner;

/**
 * A utility class for calculating discounted prices and final prices with tax.
 * 
 * <p>This class provides static methods to calculate price adjustments including
 * discount applications and tax calculations. All methods include proper input
 * validation to ensure data integrity.
 */
public final class PriceCalculator {
  
  private PriceCalculator() {
    // Utility class should not be instantiated
  }

  /**
   * Calculates the discounted price based on the original price and discount percentage.
   *
   * @param price the original price, must be non-negative
   * @param discountPercentage the discount percentage, must be between 0 and 100
   * @return the price after applying the discount
   * @throws IllegalArgumentException if discount percentage is not between 0 and 100
   */
  public static double calculateDiscountedPrice(double price, double discountPercentage) {
    if (discountPercentage < 0 || discountPercentage > 100) {
      throw new IllegalArgumentException("Discount percentage must be between 0 and 100.");
    }
    return price - (price * discountPercentage / 100.0);
  }

  /**
   * Calculates the final price after adding tax to the base price.
   *
   * @param price the base price before tax
   * @param taxRate the tax rate as a decimal (e.g., 0.05 for 5%)
   * @return the final price including tax
   * @throws IllegalArgumentException if tax rate is negative
   */
  public static double calculateFinalPriceWithTax(double price, double taxRate) {
    if (taxRate < 0) {
      throw new IllegalArgumentException("Tax rate cannot be negative.");
    }
    return price + (price * taxRate);
  }

  /**
   * Main method to demonstrate the price calculation functionality.
   *
   * @param args command line arguments (not used)
   */
  public static void main(String[] args) {
    var scanner = new Scanner(System.in);

    try {
      System.out.print("Enter the original price: ");
      var originalPrice = scanner.nextDouble();

      System.out.print("Enter the discount percentage: ");
      var discount = scanner.nextDouble();

      var discountedPrice = calculateDiscountedPrice(originalPrice, discount);
      System.out.printf("Price after %.2f%% discount: ₹%.2f%n", discount, discountedPrice);

      System.out.print("Enter the tax rate (e.g., 0.05 for 5%): ");
      var tax = scanner.nextDouble();

      var finalPrice = calculateFinalPriceWithTax(discountedPrice, tax);
      System.out.printf("Final price with tax: ₹%.2f%n", finalPrice);

    } catch (Exception e) {
      System.err.println("Error: " + e.getMessage());
    } finally {
      scanner.close();
    }
  }
}